diff --git modules/field/field.attach.inc modules/field/field.attach.inc
index 4a90961..65df71e 100644
--- modules/field/field.attach.inc
+++ modules/field/field.attach.inc
@@ -318,7 +318,11 @@ function _field_invoke_multiple($op, $entity_type, $entities, &$a = NULL, &$b =
         // Unless a language suggestion is provided we iterate on all the
         // available languages.
         $available_languages = field_available_languages($entity_type, $field);
+        // https://drupal.org/comment/8396811#comment-8396811 -JV
+        $language = (!empty($id) && !empty($options['language'][$id])) ? $options['language'][$id] : $options['language'];
+        /*
         $language = !empty($options['language'][$id]) ? $options['language'][$id] : $options['language'];
+        */
         $languages = _field_language_suggestion($available_languages, $language, $field_name);
         foreach ($languages as $langcode) {
           $grouped_items[$field_id][$langcode][$id] = isset($entity->{$field_name}[$langcode]) ? $entity->{$field_name}[$langcode] : array();
