diff --git a/theme/theme.inc b/theme/theme.inc
index 6551d7d..7d7a5ce 100644
--- a/theme/theme.inc
+++ b/theme/theme.inc
@@ -269,11 +269,11 @@ function template_preprocess_calendar_day(&$vars) {
         $group_time = NULL;
         $divisor = NULL;
         if ($display_overlap) {
-          if ($view->style_options['groupby_times'] == 'half') {
+          if ($view->style_plugin->options['groupby_times'] == 'half') {
             $group_time = 30;
             $divisor = 7.5;
           }
-          elseif ($view->style_options['groupby_times'] == 'hour') {
+          elseif ($view->style_plugin->options['groupby_times'] == 'hour') {
             $group_time = 60;
             $divisor = 15;
           }
@@ -464,6 +464,7 @@ function template_preprocess_calendar_week(&$vars) {
     }
   }

+
   // Do the headers last, once we know what the actual values are.
   $i = 0;
   $start_times = array_keys($grouped_items);
@@ -484,11 +485,11 @@ function template_preprocess_calendar_week(&$vars) {
           if ($display_overlap) {
             $group_time = NULL;
             $divisor = NULL;
-            if ($view->style_options['groupby_times'] == 'half'){
+            if ($view->style_plugin->options['groupby_times'] == 'half'){
               $group_time = 30;
               $divisor = 7.5;
             }
-            elseif ($view->style_options['groupby_times'] == 'hour'){
+            elseif ($view->style_plugin->options['groupby_times'] == 'hour'){
               $group_time = 60;
               $divisor = 15;
             }
